src/graph/sampler.cc

```diff
@@ -440,6 +441,10 @@ NodeFlow SampleSubgraph(const ImmutableGraph *graph,
                             &tmp_sampled_edge_list,
                             &time_seed);
       }
+      if (add_self_loop) {
+        tmp_sampled_src_list.push_back(dst_id);
+        tmp_sampled_edge_list.push_back(-1);
```

Comment:
Yes, but the self loop edges have no corresponding edge data. Users should not use these edge ids if they turn on this flag `add_self_loop`.

Reply:
Agree. I wonder if there is a way to tell users not to access edge data if nodefow is created with self loop

Reply:
I'll add it in the document

Reply:
If it cause segfault, it’s always bad

Reply:
I just realized a use case that might have problems with this implementation. In any heterogeneous graph, edge type is stored as edge data. After we sample a NodeFlow from such a graph, we have to use edge id to get the edge data. Since heterogeneous graphs are common graph data, we have to deal with this problem very carefully.

Reply:
could you add a check in https://github.com/dmlc/dgl/pull/322/files#diff-5bf42cb930a666d2dc57d12edcb83d59L357?